### PR TITLE
Added VAF Chart to Summary Patient View using new timeline axis

### PR DESCRIPTION
**Pull request by dianab0**
- Chart has the highlighting, tooltips and click functionality just like
    old VAF chart
    - Samples appear one under the other when having same start date
    - A "Group by" control has been added to group samples by selected
    clinical attribute (rfc 54). Samples appear on different rows depending on
    category and have different colors. Labels are not in proper place (TBD)

# What? Why?
Fix # .

Changes proposed in this pull request:
- a
- b

# Checks
- [ ] Is this PR adding logic based on one or more **clinical** attributes? If yes, please make sure validation for this attribute is also present in the data validation / data loading layers (in backend repo) and documented in [File-Formats Clinical data section](https://github.com/cBioPortal/cbioportal/blob/master/docs/File-Formats.md#clinical-data)!
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [ ] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [GifGrabber](http://www.gifgrabber.com/).

# Notify reviewers
Read our [Pull request merging
policy](../CONTRIBUTING.md#pull-request-merging-policy). If you are part of the
cBioPortal organization, notify the approprate team (remove inappropriate):

@cBioPortal/frontend

If you are not part of the cBioPortal organization look at who worked on the
file before you. Please use `git blame <filename>` to determine that
and notify them here:

### CODE REVIEW

File: packages/cbioportal-clinical-timeline/src/TimelineStore.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ import {
 import _ from 'lodash';
 import $ from 'jquery';
 import autobind from 'autobind-decorator';
+import { ClinicalEvent } from 'cbioportal-ts-api-client';
```
Comment (alisman) [inline packages/cbioportal-clinical-timeline/src/TimelineStore.ts:17]: we should probably try to keep the timeline library free of reference to other cbio libraries. lets discuss

File: packages/cbioportal-clinical-timeline/src/TimelineStore.ts [hunk 2]
```diff
@@ -49,6 +51,15 @@ export class TimelineStore {
         setTimeout(this.setScroll.bind(this), 10);
     }
 
+    setGroupByOption(value: string) {
```
Comment (alisman) [inline packages/cbioportal-clinical-timeline/src/TimelineStore.ts:54]: since this mutates observable state, it should be given @action decorator.   this is a convention which allows bulk update (when necessary). in this case, it doesn't really do anything, but still worth abiding by convention

File: packages/cbioportal-clinical-timeline/src/TimelineStore.ts [hunk 3]
```diff
@@ -49,6 +51,15 @@ export class TimelineStore {
...
+        this.groupByOption = value;
+        console.info(this.groupByOption);
+    }
+
+    getGroupByOption() {
```
Comment (alisman) [inline packages/cbioportal-clinical-timeline/src/TimelineStore.ts:59]: i would just reference this directly.   not worth wrapping in a method.

File: packages/cbioportal-clinical-timeline/src/TimelineStore.ts [hunk 4]
```diff
@@ -79,6 +90,25 @@ export class TimelineStore {
         };
     }
 
+    @autobind
```
Comment (alisman) [inline packages/cbioportal-clinical-timeline/src/TimelineStore.ts:93]: this should be a computed getter.
```
@computed get getXPositionPerSample
```

File: src/pages/patientView/SampleManager.tsx [hunk 5]
```diff
@@ -269,6 +269,16 @@ class SampleManager {
         return this.samples.map((sample: ClinicalDataBySampleId) => sample.id);
     }
 
+    getSampleIdsIndex() {
+        let indexList: { [s: string]: number } = {};
```
Comment (alisman) [inline src/pages/patientView/SampleManager.tsx:273]: this is not really a "get".    i would call it sampleIdToIndexMap and make it computed

File: src/pages/patientView/SampleManager.tsx [hunk 6]
```diff
@@ -269,6 +269,16 @@ class SampleManager {
...
+        this.samples
+            .map((sample: ClinicalDataBySampleId) => sample.id)
```
Comment (alisman) [inline src/pages/patientView/SampleManager.tsx:275]: negligible impact, but you don't need to do map.  you can just do samples.forEach(s=>indexList[s.sampleId]= ...)

Comment (alisman) [reply src/pages/patientView/SampleManager.tsx:275]: also, in general, when we transform something like this, it's best to use reduce because the code is clearest:
```const result = arr.reduce()```

File: packages/cbioportal-clinical-timeline/src/TimelineStore.ts (hunk 3 above)
Comment (adamabeshouse) [inline packages/cbioportal-clinical-timeline/src/TimelineStore.ts:59]: this could be @computed

File: src/pages/patientView/timeline2/VAFChartWrapper.tsx [hunk 7]
```diff
@@ -42,99 +96,495 @@ const VAFPointConnector: React.FunctionComponent<{
...
+        return this.props.store.getXPositionPerSample();
+    }
+
+    @computed get sampleIdOrder() {
+        return stringListToIndexSet(
+            this.props.sampleManager.getSampleIdsInOrder()
+        );
+    }
+
+    @computed get sampleGroupByColors() {
```
Comment (adamabeshouse) [inline src/pages/patientView/timeline2/VAFChartWrapper.tsx:230]: the name is confusing to me, could you please explain what this is?

File: src/pages/patientView/timeline2/VAFChartWrapper.tsx [hunk 8]
```diff
@@ -42,99 +96,495 @@ const VAFPointConnector: React.FunctionComponent<{
...
+        );
+        clinicalAttributeSamplesMap.forEach(
+            (sampleList: string[], clinicalValue: any) => {
+                groupByColors[clinicalValue] = uniqueColorGetter();
+            }
+        );
+        return groupByColors;
+    }
+
+    @computed get sampleGroupByLabels() {
```
Comment (adamabeshouse) [inline src/pages/patientView/timeline2/VAFChartWrapper.tsx:245]: could you please explain what this is?

File: src/pages/patientView/timeline2/VAFChartWrapper.tsx [hunk 9]
```diff
@@ -42,99 +96,495 @@ const VAFPointConnector: React.FunctionComponent<{
...
+        );
+        clinicalAttributeSamplesMap.forEach(
+            (sampleList: string[], clinicalValue: any) => {
+                groupByLabels.push(clinicalValue);
+            }
+        );
+        return groupByLabels;
+    }
+
+    @computed get sampleGroupByValue() {
```
Comment (adamabeshouse) [inline src/pages/patientView/timeline2/VAFChartWrapper.tsx:260]: could you please explain what this is?

File: src/pages/patientView/timeline2/VAFChartWrapper.tsx [hunk 10]
```diff
@@ -42,99 +96,495 @@ const VAFPointConnector: React.FunctionComponent<{
...
+                )!.value;
+            });
+        }
+        return groupByValue;
+    }
+
+    @computed get sampleYPosition() {
+        // compute sample y position inside the footer
+        let ystart = 0;
+        let yposition: { [s: string]: number } = {};
```
Comment (adamabeshouse) [inline src/pages/patientView/timeline2/VAFChartWrapper.tsx:283]: small thing but typically we do camelcase, so `yStart`, `yPosition`, etc

Comment (adamabeshouse) [reply src/pages/patientView/timeline2/VAFChartWrapper.tsx:283]: and lets really try to be more specific in the typing. so `{ [sampleId: string]: number }`, for example. Even though its equivalent in typescript, it makes it 10x easier to read

File: src/pages/patientView/timeline2/VAFChartWrapper.tsx [hunk 11]
```diff
@@ -42,99 +96,495 @@ const VAFPointConnector: React.FunctionComponent<{
...
+                                    const nextPoint: IPoint = data[i + 1];
+                                    if (nextPoint) {
+                                        x2 = this.xposition[nextPoint.sampleId];
+                                        y2 =
+                                            this.dataHeight -
+                                            nextPoint.y * this.dataHeight;
+                                    }
 
-export default VAFChartWrapper;
+                                    let tooltipData: { [s: string]: any } = {};
```
Comment (adamabeshouse) [inline src/pages/patientView/timeline2/VAFChartWrapper.tsx:477]: vague typing should be avoided

File: src/pages/patientView/SampleManager.tsx [hunk 12]
```diff
@@ -279,6 +289,51 @@ class SampleManager {
...
+                    clinicalAttributes.push({
+                        id: clinicalData.clinicalAttributeId,
+                        value: clinicalData.clinicalAttribute.displayName,
+                    });
+            });
+        });
+        return clinicalAttributes;
+    }
+
+    getClinicalAttributeSampleList(
```
[the remark this answers is not in the document]
Comment (dianab0) [reply src/pages/patientView/SampleManager.tsx:312]: The method returns a map between clinical attribute values and the list of sample ids associated with each value.
For example for clinical attribute "Sample Type" it returns {"Primary":["P04_Pri"], "Recurrent":["P04_Rec1", "P04_Rec2", "P04_Rec3"]. This is needed when arranging samples on different rows depending on the clinical attribute value.

File: src/pages/patientView/mutation/VAFLineChartUtils.ts [hunk 13]
```diff
@@ -20,16 +20,44 @@ function isPointBasedOnRealVAF(d: { mutationStatus: MutationStatus }) {
...
+}
+
 export function computeRenderData(
     samples: Sample[],
     mutations: Mutation[][],
     sampleIdIndex: { [sampleId: string]: number },
     mutationProfileId: string,
-    coverageInformation: CoverageInformation
+    coverageInformation: CoverageInformation,
+    groupByOption: string,
```
[the remark this answers is not in the document]
Comment (dianab0) [reply src/pages/patientView/mutation/VAFLineChartUtils.ts:52]: I get the list of clinical attributes for the Group By dropdown from sampleManager.getClinicalAttributeList as an array of clinical attribute id and clinical attribute displayName.
Once a clinical attribute is selected, the groupByOption will be set to the clinical attribute id (which is a string). But all logic is based on this id and never need to check its value.
The only value that I added and check is "None". So I can define a const GROUP_BY_NONE = "None" instead of enum, to not use the string "None" directly.

File: src/pages/patientView/mutation/VAFLineChartUtils.ts [hunk 14]
```diff
@@ -20,16 +20,44 @@ function isPointBasedOnRealVAF(d: { mutationStatus: MutationStatus }) {
...
+    sampleGroup: { [s: string]: string }
```
[the remark this answers is not in the document]
Comment (dianab0) [reply src/pages/patientView/mutation/VAFLineChartUtils.ts:53]: When user selects a clinical attribute from Group By dropdown, I need to know the group each sample is part of, in order to draw lines only between samples from same group. So this sampleGroup map is actually a map between sample Id and selected clinical attribute values. I will update the typing. A more suggestive name could be sampleIdToGroupByValueMap.

File: src/pages/patientView/timeline2/VAFChartWrapper.tsx (hunk 7 above)
Comment (dianab0) [reply src/pages/patientView/timeline2/VAFChartWrapper.tsx:230]: When user selects a clinical attribute from Group By dropdown, I need to assign a different color for each of the clinical attribute values selected in Group By. So this returns a map between clinical attribute value and color.
I wanted to keep the "Group By" in the naming as this is only for the selected option.

File: src/pages/patientView/timeline2/VAFChartWrapper.tsx (hunk 8 above)
Comment (dianab0) [reply src/pages/patientView/timeline2/VAFChartWrapper.tsx:245]: When user selects a clinical attribute from Group By dropdown, this returns the list of clinical attribute values selected. I use this as labels on the chart.

File: src/pages/patientView/timeline2/VAFChartWrapper.tsx (hunk 9 above)
Comment (dianab0) [reply src/pages/patientView/timeline2/VAFChartWrapper.tsx:260]: When user selects a clinical attribute from Group By dropdown, this returns a map between a sampleId and the associated clinical attribute value (selected in Group By). Passed as sampleGroup parameter to computeRender method.

File: src/pages/patientView/PatientViewPage.tsx [hunk 15]
```diff
@@ -202,6 +204,20 @@ export default class PatientViewPage extends React.Component<
         );
     }
 
+    private dataStore = new PatientViewMutationsDataStore(
```
Comment (alisman) [inline src/pages/patientView/PatientViewPage.tsx:207]: @alisman look at this more closely

File: src/pages/patientView/mutation/PatientViewMutationsDataStore.ts [hunk 16]
```diff
@@ -19,12 +19,19 @@ export default class PatientViewMutationsDataStore extends SimpleGetterLazyMobXT
     Mutation[]
 > {
     @observable.ref private mouseOverMutation: Readonly<Mutation> | null = null;
+    @observable.ref private mouseOverMutationTooltip: {
```
[the remark this answers is not in the document]
Comment (alisman) [reply src/pages/patientView/mutation/PatientViewMutationsDataStore.ts:22]: view specific state should be local to view components

File: packages/cbioportal-clinical-timeline/src/TimelineStore.ts (hunk 1 above)
Comment (dianab0) [reply packages/cbioportal-clinical-timeline/src/TimelineStore.ts:17]: fixed

File: packages/cbioportal-clinical-timeline/src/TimelineStore.ts (hunk 2 above)
Comment (dianab0) [reply packages/cbioportal-clinical-timeline/src/TimelineStore.ts:54]: fixed

File: packages/cbioportal-clinical-timeline/src/TimelineStore.ts (hunk 3 above)
Comment (dianab0) [reply packages/cbioportal-clinical-timeline/src/TimelineStore.ts:59]: fixed

File: packages/cbioportal-clinical-timeline/src/TimelineStore.ts (hunk 4 above)
Comment (dianab0) [reply packages/cbioportal-clinical-timeline/src/TimelineStore.ts:93]: fixed

File: src/pages/patientView/SampleManager.tsx (hunk 5 above)
Comment (dianab0) [reply src/pages/patientView/SampleManager.tsx:273]: fixed

File: src/pages/patientView/SampleManager.tsx (hunk 6 above)
Comment (dianab0) [reply src/pages/patientView/SampleManager.tsx:275]: fixed

File: packages/cbioportal-clinical-timeline/src/TimelineStore.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (dianab0) [reply packages/cbioportal-clinical-timeline/src/TimelineStore.ts:54]: fixed

File: packages/cbioportal-clinical-timeline/src/TimelineStore.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (dianab0) [reply packages/cbioportal-clinical-timeline/src/TimelineStore.ts:59]: fixed

File: packages/cbioportal-clinical-timeline/src/TimelineStore.ts [hunk 17]
```diff
@@ -79,6 +90,25 @@ export class TimelineStore {
...
+    getXPositionPerSample(): { [s: string]: number } {
```
[the remark this answers is not in the document]
Comment (dianab0) [reply packages/cbioportal-clinical-timeline/src/TimelineStore.ts:94]: fixed

File: src/pages/patientView/SampleManager.tsx [hunk 18]
```diff
@@ -279,6 +289,51 @@ class SampleManager {
...
         return '';
     }
+
+    getClinicalAttributeList(samples: Array<ClinicalDataBySampleId>) {
+        let clinicalAttributes: { id: string; value: string }[] = [];
+
+        samples.forEach((sample, sampleIndex) => {
+            sample.clinicalData.forEach((clinicalData, clinicalDataIndex) => {
+                if (
+                    clinicalAttributes.find(
```
[the remark this answers is not in the document]
Comment (dianab0) [reply src/pages/patientView/SampleManager.tsx:299]: fixed

File: src/pages/patientView/mutation/PatientViewMutationsDataStore.ts (hunk 16 above)
[the remark this answers is not in the document]
Comment (dianab0) [reply src/pages/patientView/mutation/PatientViewMutationsDataStore.ts:22]: fixed by adding the tooltipModel with more specific typing of the data;
still left the mouseOverMutation as it is used by old code (vafLineChart).

File: src/pages/patientView/timeline2/TimelineWrapper.tsx [hunk 19]
```diff
@@ -42,9 +46,10 @@ function splitCats(splits: string[], cat: any): TimelineTrack {
         };
     }
 
-    const groups = _.groupBy(
-        cat,
-        item => item.attributes.find((att: any) => att.key === splits[0]).value
+    const groups = _.groupBy(cat, item =>
+        item.attributes.find((att: any) => att.key === splits[0])
+            ? item.attributes.find((att: any) => att.key === splits[0]).value
```
[the remark this answers is not in the document]
Comment (dianab0) [reply src/pages/patientView/timeline2/TimelineWrapper.tsx:51]: fixed

File: src/pages/patientView/timeline2/VAFChartControls.tsx [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+            })),
+    ];
+
+    return (
+        <div style={{ display: 'flex', alignItems: 'center' }}>
+            <span style={{ marginTop: -3, marginRight: 3 }}>Group by:</span>
+            <div style={{ minWidth: 200, width: 200, zIndex: 20 }}>
+                <ReactSelect
+                    value={
+                        groupByOptions.find(
```
[the remark this answers is not in the document]
Comment (dianab0) [reply src/pages/patientView/timeline2/VAFChartControls.tsx:35]: fixed

File: src/pages/patientView/timeline2/VAFChartWrapper.tsx [hunk 21]
```diff
@@ -42,99 +96,495 @@ const VAFPointConnector: React.FunctionComponent<{
...
+                    hugoGeneSymbol: lineData[0].mutation.gene.hugoGeneSymbol,
+                    proteinChange: lineData[0].mutation.proteinChange,
+                },
+                lineData
+            );
+        }
+        return map;
+    }
+
+    @computed get xposition() {
```
[the remark this answers is not in the document]
Comment (dianab0) [reply src/pages/patientView/timeline2/VAFChartWrapper.tsx:220]: fixed

File: src/pages/patientView/timeline2/VAFChartWrapper.tsx (hunk 10 above)
Comment (dianab0) [reply src/pages/patientView/timeline2/VAFChartWrapper.tsx:283]: fixed

File: src/pages/patientView/timeline2/VAFChartWrapper.tsx (hunk 11 above)
Comment (dianab0) [reply src/pages/patientView/timeline2/VAFChartWrapper.tsx]: fixed

**Comment by dianab0** [comment]
Hi,
Thanks Aaron and Adam for review. I applied the changes, added also the rest of VAF controls (Show only selected mutations, Log scale and Set x-axis to data range) to the chart, and re-pushed everything as a single commit on top of latest timeline branch after solving conflicts.

File: src/pages/patientView/SampleManager.tsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (adamabeshouse) [reply src/pages/patientView/SampleManager.tsx:317]: Ahh, okay got it. So I think a better name would be `getClinicalValueToSamplesMap`

File: src/pages/patientView/mutation/PatientViewMutationsDataStore.ts (hunk 16 above)
[the remark this answers is not in the document]
Comment (adamabeshouse) [reply src/pages/patientView/mutation/PatientViewMutationsDataStore.ts]: looks good, thank you

File: src/pages/patientView/mutation/VAFLineChartUtils.ts (hunk 13 above)
[the remark this answers is not in the document]
Comment (adamabeshouse) [reply src/pages/patientView/mutation/VAFLineChartUtils.ts:52]: oh I see. how about `groupByClinicalAttributeId` ? is that right?

File: src/pages/patientView/mutation/VAFLineChartUtils.ts (hunk 14 above)
[the remark this answers is not in the document]
Comment (adamabeshouse) [reply src/pages/patientView/mutation/VAFLineChartUtils.ts:53]: okay I see. I think maybe `sampleIdToClinicalValue` would be fine, a little shorter. thanks!

File: src/pages/patientView/mutation/VAFLineChartUtils.ts [hunk 22]
```diff
@@ -20,16 +21,43 @@ function isPointBasedOnRealVAF(d: { mutationStatus: MutationStatus }) {
     );
 }
 
+function splitMutationsBySampleGroup(
```
Comment (adamabeshouse) [inline src/pages/patientView/mutation/VAFLineChartUtils.ts:24]: can you explain what this method does? I want to make a suggestion on the naming but I want to make sure I understand first what its purpose is

File: src/pages/patientView/timeline2/VAFChartWrapper.tsx (hunk 7 above)
Comment (adamabeshouse) [reply src/pages/patientView/timeline2/VAFChartWrapper.tsx:289]: ah okay. so I think a better name would be `clinicalValueToColor`

File: src/pages/patientView/timeline2/VAFChartWrapper.tsx (hunk 8 above)
Comment (adamabeshouse) [reply src/pages/patientView/timeline2/VAFChartWrapper.tsx:304]: ah okay, I think I understand your phrasing now - its "labels for the sample-group-by feature". I think that we can just go with `clinicalValuesForGrouping`

File: src/pages/patientView/timeline2/VAFChartWrapper.tsx (hunk 9 above)
Comment (adamabeshouse) [reply src/pages/patientView/timeline2/VAFChartWrapper.tsx:319]: yeah I see, I was getting really confused by the word order, but now I understand. I do think it's better to choose different wording because at first glance it's hard to parse the meaning of "sampleGroupByX" meaning "X for the sample-group-by feature". How about `sampleToClinicalValue`?

File: packages/cbioportal-clinical-timeline/src/TimelineStore.tsx [hunk 23]
```diff
@@ -31,9 +30,60 @@ export class TimelineStore {
         events: TimelineEvent[];
         index: number;
     };
+    @observable groupByOption: Readonly<string> | null = null;
```
Comment (alisman) [inline packages/cbioportal-clinical-timeline/src/TimelineStore.tsx:33]: i would like to move all this stuff out of TimelineStore, which should be generic to all timelines.   This stuff should probably be in a new store that belongs to TimelineWrapper or in component state of top level VAF.

File: src/pages/patientView/PatientViewPage.tsx [hunk 24]
```diff
@@ -447,6 +464,23 @@ export default class PatientViewPage extends React.Component<
         }
     }
 
+    @autobind
+    private onTableRowClick(d: Mutation[]) {
```
Comment (alisman) [inline src/pages/patientView/PatientViewPage.tsx:468]: we should be specific about which table this pertains to.   i.e. onMutationTableRowClick

Comment (alisman) [reply src/pages/patientView/PatientViewPage.tsx:468]: true for below as well

File: src/pages/patientView/SampleManager.tsx [hunk 25]
```diff
@@ -279,6 +292,53 @@ class SampleManager {
         }
         return '';
     }
+
+    getClinicalAttributeList(samples: Array<ClinicalDataBySampleId>) {
```
Comment (alisman) [inline src/pages/patientView/SampleManager.tsx:296]: this is essentially a static function since it does not make use of instance (this).  You could make it a helper function.   You should also improve the name:  getClinicalAttributeListForSamples

File: src/pages/patientView/SampleManager.tsx [hunk 26]
```diff
@@ -279,6 +292,53 @@ class SampleManager {
...
+                    });
+                    clinicalAttributes[clinicalData.clinicalAttributeId] =
+                        clinicalData.clinicalAttribute;
+                }
+            });
+        });
+        return output;
+    }
+
+    getClinicalValueToSamplesMap(
```
Comment (alisman) [inline src/pages/patientView/SampleManager.tsx:317]: same is true here.  it's a helper function. no mention of this

File: src/pages/patientView/SampleManager.tsx [hunk 27]
```diff
@@ -279,6 +292,53 @@ class SampleManager {
...
+        samples: Array<ClinicalDataBySampleId>,
+        clinicalAttributeId: string
+    ) {
+        let clinicalAttributeSamplesMap = new Map();
```
Comment (alisman) [inline src/pages/patientView/SampleManager.tsx:321]: @adamabeshouse, by using untyped Map() here aren't we sacrificing typing?  There's probably a way to improve with generic, right?

File: src/pages/patientView/mutation/PatientViewMutationsDataStore.ts [hunk 28]
```diff
@@ -18,7 +19,24 @@ function mutationIdKey(m: Mutation) {
 export default class PatientViewMutationsDataStore extends SimpleGetterLazyMobXTableApplicationDataStore<
     Mutation[]
 > {
-    @observable.ref private mouseOverMutation: Readonly<Mutation> | null = null;
+    @observable tooltipModel: {
```
Comment (alisman) [inline src/pages/patientView/mutation/PatientViewMutationsDataStore.ts:22]: would be preferable to put this kind of UI related state into component rather than datastore

File: src/pages/patientView/SampleManager.tsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (dianab0) [reply src/pages/patientView/SampleManager.tsx]: Fixed (changed the name)

File: src/pages/patientView/timeline2/VAFChartWrapper.tsx (hunk 7 above)
Comment (dianab0) [reply src/pages/patientView/timeline2/VAFChartWrapper.tsx]: Fixed (changed the name)

File: src/pages/patientView/timeline2/VAFChartWrapper.tsx (hunk 9 above)
Comment (dianab0) [reply src/pages/patientView/timeline2/VAFChartWrapper.tsx]: Fixed (changed the name)

File: src/pages/patientView/mutation/PatientViewMutationsDataStore.ts (hunk 28 above)
Comment (dianab0) [reply src/pages/patientView/mutation/PatientViewMutationsDataStore.ts:22]: I have moved everything to a TimelineWrapperStore. There's no code change anymore from me in PatientViewMutationsDataStore. But the mouseOverMutation in here is part of old functionality and its purpose is that when a mutation is highlighted in the VAF chart to be highlighted in the mutations table too and vice versa.  Same goes for selected mutations.

File: src/pages/patientView/PatientViewPage.tsx (hunk 24 above)
Comment (dianab0) [reply src/pages/patientView/PatientViewPage.tsx:468]: Fixed

File: src/pages/patientView/SampleManager.tsx (hunk 25 above)
Comment (dianab0) [reply src/pages/patientView/SampleManager.tsx:296]: Fixed

File: src/pages/patientView/SampleManager.tsx (hunk 26 above)
Comment (dianab0) [reply src/pages/patientView/SampleManager.tsx:317]: Fixed

File: packages/cbioportal-clinical-timeline/src/TimelineStore.tsx (hunk 23 above)
Comment (dianab0) [reply packages/cbioportal-clinical-timeline/src/TimelineStore.tsx:33]: Fixed.

File: src/pages/patientView/SampleManager.tsx [hunk 29]
```diff
@@ -279,6 +292,53 @@ class SampleManager {
...
+                    clinicalAttributes.push({
+                        id: clinicalData.clinicalAttributeId,
+                        value: clinicalData.clinicalAttribute.displayName,
+                    });
+                    clinicalAttributeIds[clinicalData.clinicalAttributeId] =
+                        clinicalData.clinicalAttribute;
+                }
+            });
+        });
+        return clinicalAttributes;
```
[the remark this answers is not in the document]
Comment (dianab0) [reply src/pages/patientView/SampleManager.tsx]: I've tried that but I have problems with the map as the index is a string, not a number. So adding the id (which is an index string) inside the map didn't work.

File: src/pages/patientView/timeline2/VAFChartWrapper.tsx (hunk 8 above)
Comment (dianab0) [reply src/pages/patientView/timeline2/VAFChartWrapper.tsx]: Fixed (changed the name)

File: src/pages/patientView/mutation/VAFLineChartUtils.ts (hunk 14 above)
[the remark this answers is not in the document]
Comment (dianab0) [reply src/pages/patientView/mutation/VAFLineChartUtils.ts]: Fixed (changed the name)

File: src/pages/patientView/mutation/VAFLineChartUtils.ts (hunk 22 above)
Comment (dianab0) [reply src/pages/patientView/mutation/VAFLineChartUtils.ts:24]: For example, when user selects "Sample type" from the Group By dropdown, then a mutation that contains all samples (P04_Pri, P04_Rec1, P04_Rec2, P04_Rec3) will be split into 2 different mutations (P04_Pri) and (P04_Rec1, P04_Rec2, P04_Rec3) as the samples are part of 2 different groups (P04_Pri is Primary while the rest are Recurrent). This is needed because the purpose of Group by is to draw lines only between the samples from same group.

File: src/pages/patientView/timeline2/TimelineWrapperStore.tsx [hunk 30]
```diff
@@ -0,0 +1,81 @@
+import { action, observable } from 'mobx';
+import _ from 'lodash';
+import * as React from 'react';
+import { Mutation } from 'cbioportal-ts-api-client';
+import { MutationStatus } from '../mutation/PatientViewMutationsTabUtils';
+
+export default class TimelineWrapperStore {
+    @observable groupByOption: Readonly<string> | null = null;
```
Comment (alisman) [inline src/pages/patientView/timeline2/TimelineWrapperStore.tsx:8]: how come you are using ReadOnly here and elsewhere?

File: src/pages/patientView/timeline2/TimelineWrapperStore.tsx [hunk 31]
```diff
@@ -0,0 +1,81 @@
...
+        datum: {
+            sampleId: string;
+            vaf: number;
+            mutationStatus: MutationStatus;
+        } | null,
+        mutation: Mutation | null,
+        mouseEvent: React.MouseEvent<any>,
+        tooltipOnPoint: boolean
+    ) {
+        this.tooltipModel.mouseEvent = mouseEvent;
```
Comment (alisman) [inline src/pages/patientView/timeline2/TimelineWrapperStore.tsx:74]: i think you are resetting all the properties of the tooltip model here.  in that case, you should consider just setting the entire object.   i.e.   this.tooltipModel = { ... } rather than updating individual properties like this.  You would then be treating it as an immutable object, which would be an optimization.   You can then make the tooltipModel   @observable.ref

File: src/pages/patientView/timeline2/TimelineWrapperStore.tsx [hunk 32]
```diff
@@ -0,0 +1,81 @@
...
+        | Readonly<boolean>
+        | undefined = undefined;
+
+    @observable vafChartLogScale: Readonly<boolean> | undefined = undefined;
+
+    @observable vafChartYAxisToDataRange:
+        | Readonly<boolean>
+        | undefined = undefined;
+
+    @observable tooltipModel: {
```
Comment (alisman) [inline src/pages/patientView/timeline2/TimelineWrapperStore.tsx:22]: unless you are really intending to update individual properties of this model (leaving others untouched) this should be @observable.ref   (see comment below)

File: src/pages/patientView/timeline2/TimelineWrapperStore.tsx (hunk 30 above)
Comment (dianab0) [reply src/pages/patientView/timeline2/TimelineWrapperStore.tsx]: Correct. It was left after a copy paste of another observable. Removed it.

File: src/pages/patientView/timeline2/TimelineWrapperStore.tsx (hunk 31 above)
Comment (dianab0) [reply src/pages/patientView/timeline2/TimelineWrapperStore.tsx]: Fixed this. Thanks

File: src/pages/patientView/timeline2/TimelineWrapperStore.tsx (hunk 32 above)
Comment (dianab0) [reply src/pages/patientView/timeline2/TimelineWrapperStore.tsx]: Fixed

**Comment by adamabeshouse** [comment]
@dianab0 can you please rebase your branch onto `master` since we've merged the timeline?

**Comment by dianab0** [comment]
This PR can be closed.
The VAF chart work has been finished under a different PR.
